### PR TITLE
expression: add field type

### PR DESCRIPTION
**Pull request by hhkbp2**
Hi,

This PR adds field type for expression. The corresponding code is in 
https://github.com/pingcap/tidb/blob/master/expression/builtin.go#L48

PTAL @XuHuaiyu @AndreMouche @lishihai9017 @andelf

### CODE REVIEW

File: proto/expression.proto [hunk 1]
```diff
@@ -11,6 +11,41 @@ option (gogoproto.marshaler_all) = true;
...
+
+	TypeTinyBlob = 0xf9;
+	TypeMediumBlob = 0xfa;
+	TypeLongBlob = 0xfb;
+	TypeBlob = 0xfc;
+	TypeVarString = 0xfd;
+	TypeString = 0xfe;
+}
+
+message FieldType {
```
Comment (hanfei1991) [inline proto/expression.proto]: Do we need the Flen and Decimal fields ?

Comment (hhkbp2) [inline proto/expression.proto]: Looks like these fields are needed, we could add them later when the code is written where these fields are used.

**Comment by lishihai9017** [comment]
LGTM

**Comment by XuHuaiyu** [comment]
LGTM

**Comment by XuHuaiyu** [comment]
PTAL @hanfei1991

**Comment by hanfei1991** [comment]
LGTM

**Pull request merged**